conans/model/conan_file.py

```diff
@@ -145,6 +146,7 @@ def __init__(self, output, runner, display_name="", user=None, channel=None):
         self._conan_using_build_profile = False
 
         self.layout = Layout()
+        self.dependencies = ConanFileDependencies()
```

Comment:
Why this new model? Trying to optimize something?

Reply:
To give access to conanfiles to their dependencies. Navigating their dependencies one conanfile should be able:
- To gather information about cpp, that is, to compute the equivalent to ``deps_cpp_info`` from every dependency ``cpp_info``
- To gather information about dependencies references, including package_ids and recipe revisions.
- To gather information about real options of dependencies. Instead of ``self.options["mydep"]``, maybe it is ``self.dependencies["mydep"].options``. 

In summary, to replace the aggregated current view ``deps_cpp_info``, ``options``, etc, to a real full "graph" view, that can be aggregated on demand only.

Reply:
Sounds good, but why it is related to this PR? shouldn't be another one with that stuff?

Reply:
The ``EnvToolchain`` needs to explore the dependencies of the conanfile. It is implementing now access via:

```python
for dep in self._conanfile.dependencies.all:
```

Either we do an aggregation of all the ``new_env`` in a ``self._conanfile.deps_new_env_info``, or we allow the conanfile to visit its dependencies and aggregate the ``new_env`` directly itself. I think this second model is more structured, clean and scalable, and this is exactly the proposal of this PR: How the environment info can be propagated from recipes to their consumers.